server/routerlicious/packages/lambdas/src/deli/lambda.ts

```diff
@@ -141,15 +142,27 @@ export class DeliLambda implements IPartitionLambda {
         this.nackMessages = lastCheckpoint.nackMessages;
 
         const msn = this.clientSeqManager.getMinimumSequenceNumber();
-        this.minimumSequenceNumber = msn === -1 ? this.sequenceNumber : msn;
+        this.noActiveClients = msn === -1;
+        this.minimumSequenceNumber = this.noActiveClients ? this.sequenceNumber : msn;
 
         this.checkpointContext = new CheckpointContext(this.tenantId, this.documentId, checkpointManager, context);
+
+        // start the idle timer when created
+        this.setIdleTimer();
     }
 
     public handler(rawMessage: IQueuedMessage) {
+        let kafkaCheckpointMessage: IQueuedMessage | undefined;
+
         // In cases where we are reprocessing messages we have already checkpointed exit early
```

Comment:
This block requires better comment now. We should explain why we need to check for noActiveClient

Reply:
I added a `getKafkaCheckpointMessage` method with a comment explaining it